unit_testing/unit_test.py

```diff
@@ -671,70 +717,70 @@
 subsec = subsec+1
+# We can load altimetry data straight from a CMEMS netcdf file on initialisation
 try:
-    nemo = coast.NEMO(dn_files + fn_nemo_dat, dn_files + fn_nemo_dom, grid_ref = 't-grid')
     altimetry = coast.ALTIMETRY(dn_files + fn_altimetry)
-    ind = altimetry.subset_indices_lonlat_box([-10,10], [45,60])
-    altimetry_nwes = altimetry.isel(t_dim=ind) #nwes = northwest europe shelf
-    crps = coast.CRPS(nemo, altimetry_nwes, 'ssh','sla_filtered', nh_radius=30)
 
-    try:
-        if len(crps.dataset.crps)==len(altimetry_nwes['sla_filtered']):
-            print(str(sec) + chr(subsec) + " OK - CRPS SONF done for every observation")
-        else:
-            print(str(sec) + chr(subsec) + " X - Problem with CRPS SONF method")
+    # Test the data has loaded using attribute comparison, as for NEMO_data
+    alt_attrs_ref = dict([('source', 'Jason-1 measurements'),
+                 ('date_created', '2019-02-20T11:20:56Z'),
+                 ('institution', 'CLS, CNES'),
+                 ('Conventions', 'CF-1.6'),])
 
-            if len(crps.crps)==len(altimetry_nwes['sla_filtered']):
-                print(str(sec) + chr(subsec) + " OK - CRPS SONF done for every observation")
-            else:
-                print(str(sec) + chr(subsec) + " X - Problem with CRPS SONF method")
-    except:
-        print(str(sec) + chr(subsec) +" FAILED")
+    # checking is LHS is a subset of RHS
+    if alt_attrs_ref.items() <= altimetry.dataset.attrs.items():
+        print(str(sec) +chr(subsec) + " OK - Altimetry data loaded: " + fn_altimetry)
+    else:
+        print(str(sec) + chr(subsec) + " X - There is an issue with loading: " + fn_altimetry)
 except:
     print(str(sec) + chr(subsec) +" FAILED")
-
+    
+'''
 #-----------------------------------------------------------------------------#
-# ( 6b ) CRPS Map Plots                                                       #
+# ( 6b ) Altimetry subsetting                                                 #
 #                                                                             #
+'''
 subsec = subsec+1
-plt.close('all')
+# The altimetry that we loaded is global so lets subset it for the North
+# West European Shelf.
 try:
-    fig, ax = crps.map_plot()
-    fig.savefig(dn_fig + 'crps_map_plot.png')
-    #plt.close(fig)
-    print(str(sec) + chr(subsec) + " OK - CRPS Map plot saved")
-except:
-    print(str(sec) + chr(subsec) + " X - CRPS Map plot not saved")
-
-#-----------------------------------------------------------------------------#
-# ( 6c ) CRPS Map Plots                                                       #
-#                                                                             #
+    ind = altimetry.subset_indices_lonlat_box([-10,10], [45,60])
+    ind = ind[::4]
+    altimetry_nwes = altimetry.isel(t_dim=ind) #nwes = northwest europe shelf
 
-plt.close('all')
-subsec = subsec+1
-try:
-    fig, ax = crps.cdf_plot(0)
-    fig.savefig(dn_fig + 'crps_cdf_plot.png')
```

Comment:
The CDF plot disappeared

Reply:
CRPS has been moved into observation objects, so their plotting routines can be used to visualise it. CDF plot's machinery still exists but I've removed the method for now (thinking of a better way to implement it).